client/src/pages/components/user/AnswerLesson.jsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import { useEffect } from "react";
+import { useCallback } from "react";
+import { useState } from "react";
+import { useParams } from "react-router";
+import LessonApi from "../../../api/LessonApi";
+import UserApi from "../../../api/UserApi";
+import { useNavigate } from "react-router";
+
+const AnswerLesson = () => {
+  if (!UserApi.isLogin()) {
+    window.location = "/";
+  }
+
+  const [page, setPage] = useState(0);
+  const [result, setResult] = useState([]);
+  const [question, setQuestion] = useState([]);
+  const [numOfQuestion, setNumOfQuestion] = useState();
+  const [selectedAnswer, setSelectedAnswer] = useState([]);
+  const [correctAnswer, setCorrectAnswer] = useState([]);
+
+  const param = useParams();
```

Comment:
```suggestion
  const { id } = useParams();
```

Use desctructuring for easier readability